### PR TITLE
CASSANDRA-17184 Add += and -= syntax to ALTER TABLE WITH

### PR DESCRIPTION
**Pull request by HadesArchitect**
**INITIAL PROPOSAL, NOT READY FOR MERGE**

Description: TBD

At the moment the only purpose of the PR is to discuss the suggested prototype implementation.

* [Jira Issue](https://issues.apache.org/jira/browse/CASSANDRA-17184)
* [Expected behavior](https://github.com/apache/cassandra/pull/1371/files#diff-60155f5fd589ec7d5d5cba38462b429d211fa0c70a3c3283972dd6e865c1eebbR265)

### CODE REVIEW

**Comment by HadesArchitect** [comment]
@blerer Here we go

File: src/java/org/apache/cassandra/cql3/statements/PropertyDefinitions.java [hunk 1]
```diff
@@ -24,57 +24,82 @@
 import org.slf4j.LoggerFactory;
 
 import org.apache.cassandra.exceptions.SyntaxException;
+import org.mortbay.io.nio.SelectorManager;
```
Comment (HadesArchitect) [inline src/java/org/apache/cassandra/cql3/statements/PropertyDefinitions.java:27]: I have no idea how did it get there... Will remove.

**Review by Gerrrr — COMMENTED** [review]
I like the overall direction this patch is aiming at - extending `Parser.g` to call a `updateProperty` instead of `addProperty` on `+=`.

As a next step, I would suggest completing the change for `ALTER KEYSPACE` and then applying the same concept to the rest of the `ALTER` statements. Some specific things you might want to focus on:

* Interaction between your changes and  `Guardrails`. For reference, see the usages of `PropertyDefinitions#updatedProperties` and `PropertyDefinitions#removeProperty`. This might become more visible when you start working on `ALTER TABLE`.
* Better test coverage. Some cases worth covering are:
  -  a combination of `+=` and `=` in a single statement
  - empty change (e.g. `+= {}`)
  - `+=` with a single value on the right side (e.g. `WITH durable_writes += true`)
  - `+=` that impacts depended parameters, e.g., switching from `NetworkTopologyStrategy` to `SimpleStrategy` and the other way round.

**Comment by HadesArchitect** [comment]
Hey @Gerrrr, thanks for the review and ideas. Sure I'll add more tests when the general approach will be approved.

**Pull request closed without merging**